test/test_files/cast/cast_to_nested_types.test

```diff
@@ -372,3 +372,44 @@ Conversion exception: Map does not allow null as key.
 -STATEMENT RETURN cast(cast([220], "decimal(21,4)[]"), "int[1]")
 ---- 1
 [220]
+
+-CASE CastToUnion
+-STATEMENT CREATE NODE TABLE Person(id INT64, description UNION(height INT64, name STRING), PRIMARY KEY(id))
+---- ok
+-STATEMENT CREATE (p:Person {id: 5, description: 6})
+---- ok
+-STATEMENT CREATE (p:Person {id: 7, description: 'Alice'})
+---- ok
+-STATEMENT MATCH (p:Person) RETURN p.*
+---- 2
+5|6
+7|Alice
+-STATEMENT RETURN CAST(127 AS UNION(a STRING, b INT64, c FLOAT))
+---- 1
+127
+-STATEMENT WITH CAST(42 AS UNION(a STRING, b INT64, c FLOAT)) AS u RETURN u, union_tag(u)
+---- 1
+42|b
+-STATEMENT WITH CAST(42 AS UNION(a STRING, c FLOAT)) AS u RETURN u, union_tag(u)
+---- 1
+42.000000|c
+-STATEMENT WITH CAST(42 AS UNION(a STRING, b BOOL)) AS u RETURN u, union_tag(u)
+---- 1
+42|a
+-STATEMENT MATCH (p:Person) RETURN CAST(p.id AS UNION(a INT64, b STRING))
+---- 2
+5
+7
+-STATEMENT MATCH (p:Person) RETURN CAST(p.id AS UNION(a FLOAT, b STRING))
+---- 2
+5.000000
+7.000000
+-STATEMENT WITH CAST(127 AS UNION(a INT64, b INT32)) AS q RETURN q, union_tag(q)
```

Comment:
Can you add more test cases?
Let's test cast every datatype to union.

Reply:
Done with https://github.com/kuzudb/kuzu/pull/5607/commits/f17d4da71ca7d00c5255ef2fa4284bffa315de9f